app/src/main/java/com/mapbox/maps/testapp/examples/terrain/SantaCatalinaActivity.kt

```diff
@@ -0,0 +1,311 @@
+          bitmap(BitmapFactory.decodeResource(resources, R.drawable.mapbox_mylocation_icon_default))
+        }
+        +image(BACKGROUND_ICON) {
+          bitmap(
+            generateBitmap(
+              this@SantaCatalinaActivity,
+              R.drawable.mapbox_mylocation_bg_shape,
+              null
+            )
+          )
+        }
+      }
+    ) { style ->
+      // hide road labels
+      style.getLayerAs<SymbolLayer>(LAYER_ROAD_ID).visibility(Visibility.NONE)
+
+      // wait for map to become idle before making request
+      mapboxMap.addOnMapChangedListener {
+        if (it == MapChange.DID_BECOME_IDLE) {
+          // execute direction request
+          executeDirectionsRequestForRoute {
+            // add a source to visualize the route
+            addSourceForRoute(style, it)
+            // animate the route
+            animateRoute(it)
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Executes a directions request.
+   * propagates the linestring geometry as high order function.
+   */
+  private fun executeDirectionsRequestForRoute(animateRoute: (LineString) -> Unit) {
+    val client = MapboxDirections.builder()
+      .origin(POINT_START)
+      .destination(POINT_END)
+      .overview(DirectionsCriteria.OVERVIEW_SIMPLIFIED)
+      .profile(DirectionsCriteria.PROFILE_WALKING)
+      .steps(true)
+      .accessToken(getString(R.string.mapbox_access_token))
+      .build()
+    client.enqueueCall(object : Callback<DirectionsResponse> {
+      override fun onResponse(
+        call: Call<DirectionsResponse>,
+        response: Response<DirectionsResponse>
+      ) {
+        animateRoute(
+          LineString.fromPolyline(
+            response.body()!!.routes()[0].geometry()!!,
+            PRECISION_6
+          )
+        )
+      }
+
+      override fun onFailure(call: Call<DirectionsResponse>, t: Throwable) {}
+    })
+  }
+
+  /**
+   * Adds a GeoJsonSource to a style to visualize a LineString
+   */
+  private fun addSourceForRoute(style: Style, lineString: LineString) {
+    style.addSource(
+      geoJsonSource(GEOJSON_SOURCE_ID) {
+        geometry(lineString)
+      }
+    )
+  }
+
+  /**
+   * Animate the route using FreeCamera API
+   */
+  private fun animateRoute(lineString: LineString) {
+    // get the overall distance of each route so we can interpolate along them
+    val routeDistance = TurfMeasurement.length(
+      lineString, TurfConstants.UNIT_KILOMETERS
+    )
+
+    // get reference to the location layer
+    val locationLayer = mapboxMap.getStyle()!!.getLayerAs<LocationIndicatorLayer>(
+      LOCATION_LAYER_ID
+    )
+
+    // cache the camera
+    val camera = mapboxMap.getFreeCameraOptions()
+
+    // get a reference to elevations of the animation
+    // note: to call getElevation the coordinate needs to be in the viewport
+    val startElevation = mapboxMap.getElevation(POINT_START)!!.toDouble()
+    val endElevation = mapboxMap.getElevation(POINT_END)!!.toDouble()
```

Comment:
nit: could avoid using `!!` and handle the `null` gracefully by providing fallback values.